nvflare/apis/app_deployer_spec.py

```diff
@@ -0,0 +1,23 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from .fl_context import FLContext
+from .workspace import Workspace
+
+
+class AppDeployerSpec(object):
+    def deploy(
+        self, workspace: Workspace, job_id: str, job_meta: dict, app_name: str, app_data: bytes, fl_ctx: FLContext
+    ) -> str:
```

Comment:
Since we are calling this "spec"
Suggest making it an abstract base class:

```
from abc import ABC, abstractmethod


class AppDeployerSpec(ABC):
    @abstractmethod
     def deploy(
        self, workspace: Workspace, job_id: str, job_meta: dict, app_name: str, app_data: bytes, fl_ctx: FLContext
    ) -> str:
    pass
```